docs/versioned_docs/version-zenith/developer-guide/python/428321-constructors.mdx

```diff
@@ -0,0 +1,108 @@
+---
+slug: /developer-guide/python/428321/constructors
+displayed_sidebar: "zenith"
+toc_max_heading_level: 2
+title: "Constructors"
+---
+
+import PartialExperimentalDocs from '../../partials/_experimental.mdx';
+
+# Constructors
+
+<PartialExperimentalDocs />
+
+The main object's constructor is exposed as the entrypoint function for a module.
+Its parameters are available as flags in the `dagger call` command directly.
+
+This is a simple way to accept module-wide configuration, or just to set
+a few attributes without having to create setter functions for them.
+
+In the Python SDK, the [`@dagger.object_type`][@object_type] decorator wraps
+[`@dataclasses.dataclass`][dataclass], which means that an `__init__()` method
+is automatically generated, with parameters that match the declared class attributes.
+
+Here is an example with an object that has typed attributes:
+
+```python file=./snippets/constructors/simple/main.py
+```
+
+And here is an example call for this module:
+
+```shell
+dagger call --name=Daggerverse message
+```
+
+The result will be:
+
+```shell
+Hello, Daggerverse!
+```
+
+Constructor arguments are documented through their attribute documentation,
+using the same annotation used on function arguments:
+
+```python file=./snippets/constructors/doc/main.py
+```
+
+The flags can be seen documented in the output of `dagger call --help`:
+
+```shell
+Flags:
+      --greeting string   The greeting to use (default "Hello")
+      --name string       Who to greet (default "World")
+
+Function Commands:
+  message       Return the greeting message
+```
+
+## Exclude argument in constructor
+
+Same as any [data class][dataclass], attributes can be excluded from the
+generated `__init__()` function, using [dataclasses.field(init=False)][field]:
+
+```python file=./snippets/constructors/exclude/main.py
+```
```

Comment:
Note that after "chaining", this brings the concept of the `dataclasses.field` descriptor, still without `dagger.field`.